pyproject.toml

```diff
@@ -183,17 +184,16 @@ skip = 'pdm.lock,docs/examples/contrib/sqlalchemy/us_state_lookup.json'
 concurrency = ["multiprocessing", "thread"]
 omit = ["*/tests/*"]
 parallel = true
+plugins = ["covdefaults"]
+source  = ["litestar"]
 
 [tool.coverage.report]
 exclude_lines = [
-  'pragma: no cover',
-  'if TYPE_CHECKING:',
-  'except ImportError.*',
-  '\.\.\.',
-  'raise NotImplementedError.*',
+  'except ImportError\b',
   'if VERSION.startswith("1"):',
   'if pydantic.VERSION.startswith("1"):',
 ]
+fail_under = 96
```

Comment:
`pytest-cov` also can do this, even if this PR won't be merged: `--cov-fail-under=96` in `addopts`

Reply:
So far, we've been relying on Sonar to manage this metric for us - can't hurt to fail earlier though if it dips below. Also, sonar reports the coverage only on new contribution code for a PR, so measuring overall at this point can be useful too.